micro-rdk/src/common/sensor.rs

```diff
@@ -51,10 +57,46 @@ pub(crate) fn register_models(registry: &mut ComponentRegistry) {
 pub type GenericReadingsResult =
     ::std::collections::HashMap<::prost::alloc::string::String, google::protobuf::Value>;
 
+#[cfg(feature = "data")]
+impl From<GenericReadingsResult> for Data {
+    fn from(value: GenericReadingsResult) -> Self {
+        Data::Struct(google::protobuf::Struct {
+            fields: HashMap::from([(
+                "readings".to_string(),
+                google::protobuf::Value {
+                    kind: Some(google::protobuf::value::Kind::StructValue(
+                        google::protobuf::Struct { fields: value },
+                    )),
+                },
+            )]),
+        })
+    }
+}
+
 pub type TypedReadingsResult<T> = ::std::collections::HashMap<String, T>;
 
 pub trait Readings {
     fn get_generic_readings(&mut self) -> Result<GenericReadingsResult, SensorError>;
+    #[cfg(feature = "data")]
+    fn get_readings_data(&mut self) -> Result<SensorData, SensorError> {
+        let reading_requested_dt = chrono::offset::Local::now().fixed_offset();
```

Comment:
Wondering a bit about whether Metadata production belongs in the leaf functions here. If `get_readings_data` returned a `sensor_data::Data`, and `Vector3::to_sensor_data` the same, could timestamp management all be lifted up to `DataCollector`? It'd let you get, for instance, the timestamp fields out of the arguments to `Vector3::to_sensor_data`, which in turn would probably let you use `From` there as well.

The downside I see to that choice would be if the metadata later was expanded in a way that could only be satisfied by the objects themselves.

What do you think?

Reply:
Originally my plan was to put this function in `DataCollector` rather than in this trait/ Then I could create `From<_> for SensorData` impls for both Vector3 and GenericSensorReadings that initially set `metadata` to None and leave it to the `DataCollector` to determine `metadata`. But the downside you mentioned came to mind and I decided against it. 

However, if you think the other design is better enough to be worth that risk, I'm happy to do that instead

Reply:
I'm leaning towards making the change. After all, the goal here is for the sensor to provide the information that goes into the `.data` `oneof`. It seems fair to conclude that the sensor shouldn't need to be involved in deciding any metadata that the data sync service might need in addition to the sensors own readings.

Also, I wonder if since `sensor_data::Data` is a real type, it wouldn't make more sense to create `From<_> for sensor_data::Data`? Then the metadata isn't even part of the story and there is no need to add any special handling around `None` for the `metadata` field.

Reply:
yeah, it's better to write conversions to `sensor_data::Data`. Ok, I'll make the change